sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -1954,6 +1954,22 @@ class Dataset[T] private[sql](
   @scala.annotation.varargs
   def agg(expr: Column, exprs: Column*): DataFrame = groupBy().agg(expr, exprs : _*)
 
+  def melt(id: String, ids: String*)
+          (value: String, values: String*): DataFrame =
+    Melt.of(this, id +: ids, value +: values)
+
+  def melt(dropNulls: Boolean, id: String, ids: String*)
+          (value: String, values: String*): DataFrame =
+    Melt.of(this, id +: ids, value +: values, dropNulls = dropNulls)
+
+  def melt(variableColumnName: String,
+           valueColumnName: String,
+           dropNulls: Boolean,
+           expr: String,
+           exprs: String*)(id: String, ids: String*): DataFrame =
```

Comment:
nit: lets be consistent with argument names

Reply:
You are right. Though, I have reworked the API, so this is gone now.